src/profile-logic/profile-data.js

```diff
@@ -1507,6 +1507,47 @@ export function isNetworkMarker(marker: TracingMarker): boolean {
   return !!(marker.data && marker.data.type === 'Network');
 }
 
+export function getNetworkMarkers(thread: Thread): Array<NetworkPayload> {
```

Comment:
I would highly recommend splitting the work of matching up start and end Network markers into a separate PR. This step requires a bit of care, and having a clear diff with a matching test would be really helpful and help give lots of confidence over the code change. Plus, this work is needed and could be used directly in the `NetworkTrack` component as well.

Reply:
:thumbsup: